client/src/components/Login.tsx

```diff
@@ -1,120 +1,99 @@
 import React, { useState, useContext } from "react";
-
-import "styles/Login.css";
-import "styles/_input.scss";
-
+import { Trans, t } from "@lingui/macro";
 import { I18n } from "@lingui/core";
 import { withI18n } from "@lingui/react";
-import { Trans, t } from "@lingui/macro";
+import classNames from "classnames";
+
+import AuthClient from "./AuthClient";
+import { JustfixUser } from "state-machine";
 import { UserContext } from "./UserContext";
+import { useInput } from "util/helpers";
 import PasswordInput from "./PasswordInput";
-import { JustfixUser } from "state-machine";
-import AuthClient from "./AuthClient";
+import EmailInput from "./EmailInput";
+import UserTypeInput from "./UserTypeInput";
 import { Alert } from "./Alert";
-import { AlertIcon, InfoIcon } from "./Icons";
+import { InfoIcon } from "./Icons";
 import Modal from "./Modal";
 
-export enum LoginState {
-  Default,
+import "styles/Login.css";
+import "styles/UserTypeInput.css";
+import "styles/_input.scss";
+
+enum Step {
+  CheckEmail,
   Login,
-  Register,
+  RegisterAccount,
+  RegisterUserType,
+  VerifyEmail,
+  VerifyEmailReminder,
 }
 
 type LoginProps = {
   i18n: I18n;
   onBuildingPage?: boolean;
   onSuccess?: (user: JustfixUser) => void;
   handleRedirect?: () => void;
+  registerInModal?: boolean;
+  setLoginRegisterInProgress?: React.Dispatch<React.SetStateAction<boolean>>;
 };
 
 const LoginWithoutI18n = (props: LoginProps) => {
-  const { i18n, onBuildingPage, onSuccess, handleRedirect } = props;
-  const userContext = useContext(UserContext);
+  const {
+    i18n,
+    onBuildingPage,
+    onSuccess,
+    handleRedirect,
+    registerInModal,
+    setLoginRegisterInProgress,
+  } = props;
 
-  const [username, setUsername] = useState("");
-  const [password, setPassword] = useState("");
-  const [header, setHeader] = useState("Log in / sign up");
-  const [subheader, setSubheader] = useState("");
-  const [loginState, setLoginState] = useState(LoginState.Default);
-  const isDefaultState = loginState === LoginState.Default;
-  const isRegisterState = loginState === LoginState.Register;
+  const userContext = useContext(UserContext);
 
+  const [showRegisterModal, setShowRegisterModal] = useState(false);
```

Comment:
should this be set to the `registerInModal` prop value?

Reply:
no, sorry it's a bit confusing. `registerInModal` is for whether it can ever be a modal (ie on the building page, rather than nav login). and `showRegisterModal` controls if it's actually open, and we only want that if they submit their email on the building page and it's not associated with a user so that it opens the modal for registration rather than staying on page for login.